com.chartboost.helium/Android~/helium-android-bridge/src/main/java/com/chartboost/heliumsdk/unity/HeliumUnityAdWrapper.kt

```diff
@@ -0,0 +1,202 @@
+                    false
+                }
+                else -> false
+            }
+        } catch (ex: Exception) {
+            Log.w(TAG, "Helium encountered an error calling Ad.readyToShow() - ${ex.message}")
+            false
+        }
+    }
+
+    fun destroy() {
+        HeliumUnityBridge.runTaskOnUiThread {
+            destroyBannerLayout()
+            ad.destroy()
+        }
+    }
+
+    private fun createBannerLayout(screenLocation: Int) {
+        if (ad !is HeliumBannerAd) {
+            return
+        }
+
+        if (activity == null)
+            return
+
+        // Create the banner layout on the given position.
+        // Check if there is an already existing banner layout. If so, remove it. Otherwise,
+        // create a new one.
+        var layout = bannerLayout
+
+        layout?.let {
+            it.removeAllViews()
+            val bannerParent = it.parent as ViewGroup
+            bannerParent.removeView(it)
+        }.also {
+            bannerLayout = RelativeLayout(activity)
+            layout = bannerLayout
+            layout?.setBackgroundColor(Color.TRANSPARENT)
+        }
+
+        /*
+            //     TopLeft = 0,
+            //     TopCenter = 1,
+            //     TopRight = 2,
+            //     Center = 3,
+            //     BottomLeft = 4,
+            //     BottomCenter = 5,
+            //     BottomRight = 6
+        */
+        var bannerGravityPosition = 0
+        when (screenLocation) {
+            0 -> bannerGravityPosition = Gravity.TOP or Gravity.LEFT
+            1 -> bannerGravityPosition = Gravity.TOP or Gravity.CENTER_HORIZONTAL
+            2 -> bannerGravityPosition = Gravity.TOP or Gravity.RIGHT
+            3 -> bannerGravityPosition = Gravity.CENTER_VERTICAL or Gravity.CENTER_HORIZONTAL
+            4 -> bannerGravityPosition = Gravity.BOTTOM or Gravity.LEFT
+            5 -> bannerGravityPosition = Gravity.BOTTOM or Gravity.CENTER_HORIZONTAL
+            6 -> bannerGravityPosition = Gravity.BOTTOM or Gravity.RIGHT
+        }
+        layout?.gravity = bannerGravityPosition
+
+        // Attach the banner layout to the activity.
+        val density = displayDensity
+        try {
+            when (ad.getSize() ?: HeliumBannerSize.STANDARD) {
+                HeliumBannerSize.LEADERBOARD -> ad.layoutParams = getBannerLayoutParams(density, LEADERBOARD.first, LEADERBOARD.second)
+                HeliumBannerSize.MEDIUM -> ad.layoutParams = getBannerLayoutParams(density, MEDIUM.first, MEDIUM.second)
+                HeliumBannerSize.STANDARD -> ad.layoutParams = getBannerLayoutParams(density, STANDARD.first, STANDARD.second)
+            }
+
+            // Attach the banner to the banner layout.
+            layout?.addView(ad)
+            activity.addContentView(layout, ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT))
```

Comment:
What if `layout` is `null`? Do we want to return or log something or throw an error?

Reply:
This has been corrected.